File: sdk/php/src/Model/ClientOption.php
<?php

namespace KuCoin\UniversalSDK\Model;

/**
 * Holds configuration details for authentication and endpoints.
 */
class ClientOption
{
    /**
     * Authentication key for the client.
     * @var string
     */
    public $key;

    /**
     * Authentication secret for the client.
     * @var string
     */
    public $secret;

    /**
     * Authentication passphrase for the client.
     * @var string
     */
    public $passphrase;

    /**
     * The name of the broker.
     * @var string
     */
    public $brokerName;

    /**
     * The partner associated with the broker.
     * @var string
     */
    public $brokerPartner;

    /**
     * The secret key for the broker.
     * @var string
     */
    public $brokerKey;

    /**
     * Spot market API endpoint for the client.
     * @var string
     */
    public $spotEndpoint;

    /**
     * Futures market API endpoint for the client.
     * @var string
     */
    public $futuresEndpoint;

    /**
     * Broker API endpoint for the client.
     * @var string
     */
    public $brokerEndpoint;

    /**
     * Configuration for HTTP transport.
     * @var TransportOption|null
     */
    public $transportOption;

    /**
     * Configuration for WebSocket transport.
     * @var WebSocketClientOption|null
     */
    public $websocketClientOption;

    public function __construct(
        $key,
        $secret,
        $passphrase,
        $spotEndpoint,
        $futuresEndpoint,
        $brokerEndpoint,
        $brokerName,
        $brokerPartner,
        $brokerKey,
        $transportOption = null,
        $websocketClientOption = null
    )
    {
        $this->key = $key;
        $this->secret = $secret;
        $this->passphrase = $passphrase;
        $this->spotEndpoint = $spotEndpoint;
        $this->futuresEndpoint = $futuresEndpoint;
        $this->brokerEndpoint = $brokerEndpoint;
        $this->brokerName = $brokerName;
        $this->brokerPartner = $brokerPartner;
        $this->brokerKey = $brokerKey;
        $this->transportOption = $transportOption;
        $this->websocketClientOption = $websocketClientOption;
    }
}

File: sdk/java/src/main/java/com/kucoin/universal/sdk/model/ClientOption.java
package com.kucoin.universal.sdk.model;

import lombok.Builder;
import lombok.Getter;
import lombok.ToString;

/**
 * ClientOption holds the configuration details for a client, including authentication keys, API
 * endpoints, and transport options.
 */
@ToString
@Getter
@Builder(builderClassName = "Builder")
public class ClientOption {

  /* ---------- required auth ---------- */
  private final String key;
  private final String secret;
  private final String passphrase;

  /* ---------- optional broker ---------- */
  private final String brokerName;
  private final String brokerPartner;
  private final String brokerKey;

  /* ---------- endpoints ---------- */
  private final String spotEndpoint;
  private final String futuresEndpoint;
  private final String brokerEndpoint;

  /* ---------- transport tuning ---------- */
  private final TransportOption transportOption;
  private final WebSocketClientOption websocketClientOption;
}
